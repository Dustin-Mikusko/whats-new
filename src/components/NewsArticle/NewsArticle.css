.article {
  display: flex;
  flex-direction: column;
  text-align: center;
  width: 450px;
  height: 65vh;;
  background-color: white;
  margin: auto;
  margin-bottom: 25px;
  position: relative;
}

.article-title {
  font-weight: bold;
}

img {
  height: 30vh;
  width: 100%;
  margin-bottom: 15px;
}

.article-content {
  padding: 15px;
  text-align: center;
}

a {
  background-color: rgb(176, 63, 208);
  position: absolute;
  bottom: 0;
  text-align: center;
  width: 100%;
  height: 5vh;
  color: white;
  text-decoration: none;
  font-size: 1.5em;
}

.link {
  display: flex;
  justify-content: space-between;
}
