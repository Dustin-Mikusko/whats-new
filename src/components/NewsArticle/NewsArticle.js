import React from 'react';
import './NewsArticle.css';

const NewsArticle = ({ title, description, img, url }) => {
  return (
    <article className="article">
      <img alt="article snapshot"src={img}></img>
      <div className="article-content">
        <h2 className="article-title">{title}</h2>
        <p>{description}</p>
      </div>
      <div className="link">
        <a href={url} rel="noopener noreferrer" target="_blank">View Article</a>
      </div>
    </article>
  )
}

export default NewsArticle;
