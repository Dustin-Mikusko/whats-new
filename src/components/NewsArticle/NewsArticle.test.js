import React from 'react';
import NewsArticle from './NewsArticle';
import { shallow } from 'enzyme';

describe('NewsArticle', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<NewsArticle 
      title='Headline Title'
      description='Article description'
      img='url to image'
      url='link to article url'
      />
    );
  });

  it('should match the snapshot with all the prop data passed correctly', () => {
    expect(wrapper).toMatchSnapshot();
  })
})
