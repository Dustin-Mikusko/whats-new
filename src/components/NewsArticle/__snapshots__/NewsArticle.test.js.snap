// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsArticle should match the snapshot with all the prop data passed correctly 1`] = `
<article
  className="article"
>
  <img
    alt="article snapshot"
    src="url to image"
  />
  <div
    className="article-content"
  >
    <h2
      className="article-title"
    >
      Headline Title
    </h2>
    <p>
      Article description
    </p>
  </div>
  <div
    className="link"
  >
    <a
      href="link to article url"
      rel="noopener noreferrer"
      target="_blank"
    >
      View Article
    </a>
  </div>
</article>
`;
